### PR TITLE
SaaSquatch Docs Update

### PR DESCRIPTION
**Pull request by loganvolkers**
## Description of the change
-Updated `Meta.tsx`
-   Change title slug to "SaaSquatch Docs"
-   Add SEO Content

-Fix 404 errors 
-Edit `package-lock.json ` to run locally

## Types of changes

- [ ] Documentation content
- [x] Page Layout / templates / structure
- [x] Internal / code cleanup / build system

### CODE REVIEW

File: package-lock-og.json [hunk 1]
```diff
@@ -0,0 +1,13304 @@
+{
+    "name": "saasquatch-docs",
+    "requires": true,
```
Comment (DerekSiemens) [inline package-lock-og.json]: This needs to be deleted

**Comment by DerekSiemens** [comment]
The package locks that you added @zachserviss need to be deleted and removed from this PR

**Comment by DerekSiemens** [comment]
Title and description need to be updated for the changes made in this PR

**Comment by zachserviss** [comment]
Changes:
-Updated `Meta.tsx`
-   Change title slug to "SaaSquatch Docs"
-   Add SEO Content

-Fix 404 errors 
-Edit `package-lock.json ` to run locally

**Comment by DerekSiemens** [comment]
@zachserviss That should go into the section at the top of the PR

File: src/components/Meta.tsx [hunk 2]
```diff
@@ -23,43 +23,47 @@ export default function render(
...
+    if(props.fields.ogFeaturedImage){
+      ogImageURL = props?.fields?.ogFeaturedImage[0]?.fields?.file?.url;
+    }
+  }else{
+    ogImageURL = null;
+  }
+  
   const plainHighlights = striptags(markdown(props.highlights));
   return (
+    console.log(props),
```
Comment (DerekSiemens) [inline src/components/Meta.tsx]: Flagging console log to be removed

File: src/components/Meta.tsx [hunk 3]
```diff
@@ -37,29 +45,23 @@ export default function render(
           "docs sectionType-" + props.sectionType + " " + props.category
         }
       />
+      {/* SEO content */}
+      <meta name="description" content={props.fields?.seoDescription}/>
```
Comment (DerekSiemens) [inline src/components/Meta.tsx]: Why not use props.highlights as well here as we did before? I think this is needed for the historical MD documents, please investigate @zachserviss

File: src/components/Meta.tsx [hunk 4]
```diff
@@ -37,29 +45,23 @@ export default function render(
...
+        
+      <meta property="og:image" content={ogImageURL} />
+      <meta name="twitter:image" content="image url from Contentful response" />
+      
+      <meta name="robots" content={props.fields?.robotsTag} />
+      <link rel="canonical" href={props.fields?.canonicalUrl}/>
+      
+      <script type="application/ld+json">
+      {/* // JSON from Contentful */}
```
Comment (DerekSiemens) [inline src/components/Meta.tsx]: Add some more comments here about what this is for and how it connects to contentful in the future

File: src/components/Meta.tsx [hunk 5]
```diff
@@ -23,11 +23,19 @@ export default function render(
...
+    title = props.title + " | SaaSquatch " + (props.categoryName || "Docs");
   }else{
-    title = "SaaSquatch Help Center";
+    title = "SaaSquatch Docs";
   }
-    
+  
+  let ogImageURL:string;
+  if(props?.fields?.ogFeaturedImage == null){
+    ogImageURL = null;
```
Comment (DerekSiemens) [inline src/components/Meta.tsx]: Setting this to null feels a bit off, I think there is a cleaner way to achieve whats happening in this code block

File: src/components/Meta.tsx [hunk 6]
```diff
@@ -23,11 +23,17 @@ export default function render(
...
-    title = props.title + " | SaaSquatch " + (props.categoryName || "Documentation");
+    title = props.title + " | SaaSquatch " + (props.categoryName || "Docs");
   }else{
-    title = "SaaSquatch Help Center";
+    title = "SaaSquatch Docs";
   }
-    
+  
+  let ogImageURL:string;
+  if(props?.fields?.ogFeaturedImage != null){
```
Comment (DerekSiemens) [inline src/components/Meta.tsx]: Need to check the length of the array as well

File: src/components/Meta.tsx [hunk 7]
```diff
@@ -37,29 +43,26 @@ export default function render(
           "docs sectionType-" + props.sectionType + " " + props.category
         }
       />
+      {/* SEO content */}
+      <meta name="description" content={props.fields?.seoDescription || props.highlights}/>
+        
+      <meta property="og:image" content={ogImageURL} />
+      <meta name="twitter:image" content="image url from Contentful response" />
```
Comment (DerekSiemens) [inline src/components/Meta.tsx]: re-use ogImageURL?

File: src/components/Meta.tsx [hunk 8]
```diff
@@ -37,29 +43,26 @@ export default function render(
...
+      
+      <meta name="robots" content={props.fields?.robotsTag} />
+      <link rel="canonical" href={props.fields?.canonicalUrl}/>
+      
+      {/*This is part of the SEO fields (https://www.smartercontent.info/amp/6-fields-that-should-be-in-every-contentful-seo-component),  
+        Need contenful feild to diaplay this information.
```
Comment (DerekSiemens) [inline src/components/Meta.tsx]: flagging typo

File: src/components/Meta.tsx [hunk 9]
```diff
@@ -37,29 +43,26 @@ export default function render(
...
+      <meta property="og:image" content={ogImageURL} />
+      <meta name="twitter:image" content="image url from Contentful response" />
+      
+      <meta name="robots" content={props.fields?.robotsTag} />
+      <link rel="canonical" href={props.fields?.canonicalUrl}/>
+      
+      {/*This is part of the SEO fields (https://www.smartercontent.info/amp/6-fields-that-should-be-in-every-contentful-seo-component),  
+        Need contenful feild to diaplay this information. 
+      */}
+      <script type="application/ld+json">
```
Comment (DerekSiemens) [inline src/components/Meta.tsx:58]: comment out script for now

Comment (DerekSiemens) [inline src/components/Meta.tsx]: Remove for now

File: src/containers/single/issues.tsx [hunk 10]
```diff
@@ -36,7 +37,20 @@ export default function render() {
               return (
                 <tr key={issue.title}>
                   <td>
-                    <Link to={anchor}>{issue.title}</Link>
+                    <Link to={anchor}>
+                     
+                      
+                    {/* {(() => {
+                      if (issue.title == 'rs017'){
```
Comment (DerekSiemens) [inline src/containers/single/issues.tsx]: Cleanup needed

File: src/components/Meta.tsx [hunk 11]
```diff
@@ -37,29 +43,26 @@ export default function render(
           "docs sectionType-" + props.sectionType + " " + props.category
         }
       />
+      {/* SEO content */}
+      <meta name="description" content={props.fields?.seoDescription || props.highlights}/>
+        
+      <meta property="og:image" content={ogImageURL} />
```
Comment (DerekSiemens) [inline src/components/Meta.tsx]: We actually should have a default that should be overridable via the field in contentful, same for twitter

File: src/components/TocFrame.tsx [hunk 12]
```diff
@@ -20,7 +20,20 @@ export type Entry = {
   category?: string;
   categoryName?: string;
   sectionType?: string;
-  fields?: { seoDescription?: string };
+  fields?: { 
+    seoDescription?: string,
+    ogFeaturedImage?: {
```
Comment (DerekSiemens) [inline src/components/TocFrame.tsx]: Flagging type fix needed, you mentioned this was actually an array of objects

File: src/components/Meta.tsx [hunk 13]
```diff
@@ -23,11 +23,17 @@ export default function render(
...
   if(props.title){
-    title = props.title + " | SaaSquatch " + (props.categoryName || "Documentation");
+    title = props.title + " | SaaSquatch " + (props.categoryName || "Docs");
   }else{
-    title = "SaaSquatch Help Center";
+    title = "SaaSquatch Docs";
+  }
+  
+  let ogImageURL:string;
+  if(props?.fields?.ogFeaturedImage != null && Object.keys(props?.fields?.ogFeaturedImage).length >0){
```
Comment (DerekSiemens) [inline src/components/Meta.tsx]:
```suggestion
  if(props?.fields?.ogFeaturedImage != null && props?.fields?.ogFeaturedImage?.length > 0){
```

**Pull request closed without merging**